src/main/java/com/github/rnewson/couchdb/lucene/util/Analyzers.java

```diff
@@ -186,13 +280,313 @@ protected TokenStreamComponents wrapComponents(String fieldName, TokenStreamComp
         }
     }
 
+	public static Analyzer fromSpec(final JSONObject json, final String analyzerKey) throws JSONException {
+		JSONObject spec = json.optJSONObject(analyzerKey);
+		if (spec != null) {
+			return getAnalyzer(spec);
+		} else {
+			return getAnalyzer(json.optString(analyzerKey, Constants.DEFAULT_ANALYZER));
+		}
+	}
+
+	public static Analyzer fromSpec(final JSONObject json) throws JSONException {
+		return fromSpec(json, Constants.ANALYZER);
+	}
+
+	/*
+	 * called from DatabaseIndexer when handling an http search request
+	 */
+	public static Analyzer fromSpec(String str) throws JSONException {
+		if (str == null) {
+			return getAnalyzer(Constants.DEFAULT_ANALYZER);
+		} 
+		
+		str = str.trim();
+		
+		if (str.startsWith("{")) {
+			try {
+				return getAnalyzer( new JSONObject(str) );
+			
+			} catch (Exception ex) {
```

Comment:
catch Exception is quite broad, what is really thrown here that we should catch?